pkg/distro/fedora/imagetype.go

```diff
@@ -418,10 +418,25 @@ func (t *imageType) checkOptions(bp *blueprint.Blueprint, options distro.ImageOp
 		return []string{w}, nil
 	}
 
-	if customizations.GetInstaller() != nil {
+	instCust, err := customizations.GetInstaller()
+	if err != nil {
+		return nil, err
+	}
+	if instCust != nil {
 		// only supported by the Anaconda installer
 		if slices.Index([]string{"iot-installer"}, t.name) == -1 {
-			return nil, fmt.Errorf("installer customizations are not supported for %q", t.name)
+			return nil, fmt.Errorf("installer customizations are not supported for %q", t.Name())
+		}
+
+		// NOTE: the image type check is redundant with the check above, but
+		// let's keep it explicit in case one of the two changes.
+		// The kickstart contents is incompatible with the users and groups
+		// customization only for the iot-installer.
+		if t.Name() == "iot-installer" &&
+			instCust.Kickstart != nil &&
```

Comment:
This check here (and the other similar ones) seem to reimplement kickstart.Options.Validate() (in slightly different words) (also in customization.GetInstaller() similar validation is done) - I wonder if we can consolidate the validation maybe into a single place ideally? I poked around a little bit and one way would be to just use `kickstart.New().Validate()` here which seems to be equivalent to "GetInstaller()" (needs some small tweaks to kickstart.New() that are probably useful anyway, i.e. return nil if nothing is to kickstart but this might also be dangerous, needs some more thought).

Something like:
```go
diff --git a/pkg/customizations/kickstart/kickstart.go b/pkg/customizations/kickstart/kickstart.go
index 4b5897d59..6a92d0baa 100644
--- a/pkg/customizations/kickstart/kickstart.go
+++ b/pkg/customizations/kickstart/kickstart.go
@@ -52,23 +52,23 @@ type Options struct {
 }
 
 func New(customizations *blueprint.Customizations) (*Options, error) {
-       options := &Options{
-               Users:  users.UsersFromBP(customizations.GetUsers()),
-               Groups: users.GroupsFromBP(customizations.GetGroups()),
-       }
-
        instCust, err := customizations.GetInstaller()
        if err != nil {
                return nil, err
        }
-       if instCust != nil {
-               options.SudoNopasswd = instCust.SudoNopasswd
-               options.Unattended = instCust.Unattended
-               if instCust.Kickstart != nil {
-                       options.UserFile = &File{Contents: instCust.Kickstart.Contents}
-               }
+       if instCust == nil {
+               return nil, nil
        }
 
+       options := &Options{
+               Users:  users.UsersFromBP(customizations.GetUsers()),
+               Groups: users.GroupsFromBP(customizations.GetGroups()),
+       }
+       options.SudoNopasswd = instCust.SudoNopasswd
+       options.Unattended = instCust.Unattended
+       if instCust.Kickstart != nil {
+               options.UserFile = &File{Contents: instCust.Kickstart.Contents}
+       }
        if err := options.Validate(); err != nil {
                return nil, err
        }
diff --git a/pkg/distro/fedora/imagetype.go b/pkg/distro/fedora/imagetype.go
index ba1cc926c..4244b9835 100644
--- a/pkg/distro/fedora/imagetype.go
+++ b/pkg/distro/fedora/imagetype.go
@@ -10,6 +10,7 @@ import (
        "github.com/osbuild/images/internal/workload"
        "github.com/osbuild/images/pkg/blueprint"
        "github.com/osbuild/images/pkg/container"
+       "github.com/osbuild/images/pkg/customizations/kickstart"
        "github.com/osbuild/images/pkg/customizations/oscap"
        "github.com/osbuild/images/pkg/disk"
        "github.com/osbuild/images/pkg/distro"
@@ -418,25 +419,20 @@ func (t *imageType) checkOptions(bp *blueprint.Blueprint, options distro.ImageOp
                return []string{w}, nil
        }
 
-       instCust, err := customizations.GetInstaller()
+       ks, err := kickstart.New(customizations)
        if err != nil {
                return nil, err
        }
-       if instCust != nil {
+       if ks != nil {
                // only supported by the Anaconda installer
                if slices.Index([]string{"iot-installer"}, t.name) == -1 {
                        return nil, fmt.Errorf("installer customizations are not supported for %q", t.Name())
                }
 
-               // NOTE: the image type check is redundant with the check above, but
-               // let's keep it explicit in case one of the two changes.
-               // The kickstart contents is incompatible with the users and groups
-               // customization only for the iot-installer.
-               if t.Name() == "iot-installer" &&
-                       instCust.Kickstart != nil &&
-                       len(instCust.Kickstart.Contents) > 0 &&
-                       (customizations.GetUsers() != nil || customizations.GetGroups() != nil) {
-                       return nil, fmt.Errorf("iot-installer installer.kickstart.contents are not supported in combination with users or groups")
+               if t.Name() == "iot-installer" {
+                       if err := ks.Validate(); err != nil {
+                               return nil, err
+                       }
                }
        }
````
(and keep the comment, my diff is bad for this) - here and in the various other cases.

But the downside of this approach is that the API is less symetric now, "customizations.GetInstaller()" seems to be not really needed anymore with this change and that is not too nice in the context of the other customizations.Get* - anyway, rambling but maybe something interesting is in this :)
```